Userland/Libraries/LibWeb/CSS/StyleValue.h

```diff
@@ -340,6 +347,82 @@ class LengthStyleValue : public StyleValue {
     Length m_length;
 };
 
+class CalculatedStyleValue : public StyleValue {
+public:
+    struct CalcSum;
+    struct CalcSumPartWithOperator;
+    struct CalcProduct;
+    struct CalcProductPartWithOperator;
+
+    // FIXME: These aren't used. They are needed for nested calc()'s
+    struct CalcNumberSum {
+    };
+    struct CalcNumberProduct {
+    };
+    struct CalcNumberProductWithOperator {
+    };
+    struct CalcNumberValueWithOperator {
+    };
+
+    using CalcNumberValue = Variant<float, CalcNumberSum>;
+    using CalcValue = Variant<float, CSS::Length, CalcSum>;
+
+    // This represents that: https://drafts.csswg.org/css-values-3/#calc-syntax
+    struct CalcSum {
+        NonnullOwnPtr<CalcProduct> first_calc_product;
+        NonnullOwnPtrVector<CalcSumPartWithOperator> zero_or_more_additional_calc_products;
+
+        CalcSum(NonnullOwnPtr<CalcProduct> first_calc_product_, NonnullOwnPtrVector<CalcSumPartWithOperator> additional)
+            : first_calc_product(move(first_calc_product_))
+            , zero_or_more_additional_calc_products(move(additional)) {};
+    };
+
+    struct CalcProduct {
+        CalcValue first_calc_value;
+        NonnullOwnPtrVector<CalcProductPartWithOperator> zero_or_more_additional_calc_values;
+    };
+
+    struct CalcSumPartWithOperator {
+        enum Operation {
+            Add,
+            Subtract,
+        } op;
+        NonnullOwnPtr<CalcProduct> calc_product;
+
+        CalcSumPartWithOperator(Operation op_, NonnullOwnPtr<CalcProduct> calc_product_)
+            : op(op_)
+            , calc_product(move(calc_product_)) {};
+    };
+
+    struct CalcProductPartWithOperator {
+        enum {
+            Multiply,
+            Divide,
+        } op;
+        Variant<CalcValue, CalcNumberValue> value {};
```

Comment:
I'm not sure what this is constructing as (probably a default-constructed `CalcNumberValue` which contains a `CalcNumberSum`), but it's almost definitely not what you expect (empty?)
For that, explicitly add `Empty` as an alternative and construct it with `Empty { }`
But if that _is_ what you expect it to do, please do so explicitly.